lib/cfg/cfg-parsers/llvm-ir.ts

```diff
@@ -62,7 +62,7 @@ export class LlvmIrCfgParser extends BaseCFGParser {
         while (i < asmArr.length) {
             if (this.functionDefinition.test(asmArr[i].text)) {
                 const start = i;
-                while (asmArr.length > 0 && asmArr[i].text !== '}') {
+                while (i < asmArr.length - 1 && asmArr[i].text !== '}') {
```

Comment:
This turned out to NOT be the root cause, but is an obvious bug nevertheless.

Reply:
not `i < asmArr.length && etc` ? why -1?

Reply:
`i` will never be `asmArr.length` if it's `<`

Reply:
```ts
while (i < asmArr.length && asmArr[i].text !== '}') {
// suppose i == asmArr.length-1 here
    i++;
// now i is asmArr.length
}

result.push({
    start,
    end: i,   // and this is the overflowed value pushed to result.
});
```
I think..

Reply:
Maybe it would be easier to change this to a do-while loop, I don't see a situation where `i++` doesn't happen at all.

Reply:
ahhhhh right
if the last character is not `}`, `i` will be `asmArr.length - 1 + 1`
confusing code, but whatever

Reply:
Yeah I'll change it to do-while. Gimme a minute